contribute/design/1.x/csi/20190805-csi-zfspv-volume-provisioning.md

```diff
@@ -261,6 +258,123 @@ When CSI will get volume destroy request, it will destroy the created zvol and a
 ### 4. CSI ZFSPV property change
 There will be a watcher watching for this ZFSVolume custom resource in the agent. We can update the ZFSVolume custom resource with the desired property and the watcher of this custom resource will apply the changes to the corresponding volume.
 
+### 5. CSI SnapShot
+
+When ZFS CSI controller gets a snapshot create request :-
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: zfspv-snap
+  namespace: openebs
+spec:
+  volumeSnapshotClassName: zfspv-snapclass
+  source:
+    persistentVolumeClaimName: zfspv-pvc
+```
+
+it will create a ZFSSnapshot custom resource. This custom resuorce will get all the details from the ZFSVolume CR, ownerNodeID will be same here as snapshot will also be there on same node where the original volume is there. The controller will create the request with status as Pending and set a label "openebs.io/volname" to the volume name from where the snapshot has to be created.
+
+```yaml
+apiVersion: openebs.io/v1alpha1
+kind: ZFSSnapshot
+metadata:
+  name: zfspv-snap
+  namespace: openebs
+  labels:
+    openebs.io/volname: pvc-34133838-0d0d-11ea-96e3-42010a800114
```

Comment:
adding the details of the volumesnapshot that created this snap will be useful. 
Also let us keep the labels uniform with other engines - openebs.io/persistent-volume.

Reply:
updated the label. Volumesnapshot yaml is already there.